src/components/Steps/TestDetails.tsx

```diff
@@ -1,117 +1,119 @@
-            instanceId="test_typeSelect"
-            isSearchable
             placeholder="Test Type"
           />
-          <Select
-            className={styles.custom_input}
+          <SelectField
+            control={control}
+            name_="format"
             options={TestFormatOptions}
-            value={selectedTestFormat}
-            onChange={(selectedOption) => setSelectedTestFormat(selectedOption)}
-            instanceId="test_formatSelect"
-            isSearchable
             placeholder="Test Format"
           />
-          <Select
-            className={styles.custom_input}
+          <SelectField
+            control={control}
+            name_="purpose"
             options={TestPurposeOptions}
-            value={selectedTestPurpose}
-            onChange={(selectedOption) =>
-              setSelectedTestPurpose(selectedOption)
-            }
-            instanceId="test_purposeSelect"
-            isSearchable
             placeholder="Test Purpose"
           />
-          <Select
-            className={styles.custom_input}
+          <SelectField
+            control={control}
+            name_="platform"
             options={TestPlatformOptions}
-            value={selectedTestPlatform}
-            onChange={(selectedOption) =>
-              setSelectedTestPlatform(selectedOption)
-            }
-            instanceId="test_platformSelect"
-            isSearchable
             placeholder="Test Platform"
           />
-          <Select
-            className={styles.custom_input}
+          <SelectField
+            control={control}
+            name_="markingScheme"
+            options={MarkingSchemeOptions}
+            placeholder="Test Purpose"
+          />
+          <SelectField
+            control={control}
+            name_="optionalLimit"
             options={OptionalLimitOptions}
-            value={selectedOptionalLimit}
-            onChange={(selectedOption) =>
-              setSelectedOptionalLimit(selectedOption)
-            }
-            instanceId="Optional_limitSelect"
-            isSearchable
             placeholder="Optional Limit"
           />
 
-          <Select
+          <input
+            required
             className={styles.custom_input}
-            options={MarkingSchemeOptions}
-            value={selectedMarkingScheme}
-            onChange={(selectedOption) =>
-              setSelectedMarkingScheme(selectedOption)
-            }
-            instanceId="marking_schemeSelect"
-            isSearchable
-            placeholder="Marking Scheme"
+            placeholder="Test id"
+            {...register("id")}
           />
-          <input className={styles.custom_input} placeholder="Test id" />
           <input
+            required
             className={styles.custom_input}
             placeholder="Test Session id"
+            {...register("sessionId")}
           />
           <input
+            required
```

Comment:
why set it as `required` -- the idea was to remove them from form na

Reply:
oh yes , removed them but should we have a auth layer so that it is only available to a certain set of users?

Reply:
ya i think there'll be an auth layer for quiz-creator "writing" part. but reading part is open to all.. at least in AF. 
thing is -- links get generated after a while (atleast a minute if automated, or more than few hours if manual run to generate links). so, session_id and quiz_id etc will remain empty for a while